### PR TITLE
Replace FilterArgument middleware with Filtering

### PR DESCRIPTION
**Pull request by progsmile**
#3873

### CODE REVIEW

File: apps/graphql/lib/graphql_web/loaders/prm.ex [hunk 1]
```diff
@@ -39,9 +39,5 @@ defmodule GraphQLWeb.Loaders.PRM do
 
   def query(queryable, _), do: queryable
 
-  defp filter(query, [{:merged_from_legal_entity, filter} | tail]) do
-    filter(query, [{:merged_from, filter} | tail])
-  end
-
-  defp filter(query, filter), do: Search.filter(query, filter)
+  defp filter(query, filter), do: Filtering.filter(query, filter)
```
Comment (PavelVesnin) [inline apps/graphql/lib/graphql_web/loaders/prm.ex]: why not to import `Filtering.filter/2`?

File: apps/graphql/lib/graphql_web/schema/division_types.ex [hunk 2]
```diff
@@ -48,7 +48,7 @@ defmodule GraphQLWeb.Schema.DivisionTypes do
 
   input_object :division_filter do
     field(:database_id, :id)
-    field(:name, :string_like)
+    field(:name, :string)
```
Comment (PavelVesnin) [inline apps/graphql/lib/graphql_web/schema/division_types.ex]: 👍

File: apps/graphql/lib/graphql_web/loaders/prm.ex (hunk 1 above)
Comment (progsmile) [reply apps/graphql/lib/graphql_web/loaders/prm.ex]: Remember what are cases to import?

Comment (PavelVesnin) [reply apps/graphql/lib/graphql_web/loaders/prm.ex]: No. What cases?

Comment (progsmile) [reply apps/graphql/lib/graphql_web/loaders/prm.ex]: Import infrastructure functions, aliasing business, so here should be import (but personally I would aliasing everything)

Comment (progsmile) [reply apps/graphql/lib/graphql_web/loaders/prm.ex]: oh, ok, I've got the point

**Pull request merged**